tensorflow/compiler/xla/service/gpu/tree_reduction_rewriter.cc

```diff
@@ -97,9 +97,27 @@ class ReductionRewriterVisitor : public DfsHloRewriteVisitor {
                                                  input_shape, input);
     }
     bool is_row_reduction = reduction_dimensions.is_row_reduction;
-
-    int64 atomic_free_bound =
-        is_row_reduction ? kRowAtomicFreeBound : kColumnAtomicFreeBound;
+    int64 atomic_free_bound = 0;
+    if (is_row_reduction) {
```

Comment:
Please add a comment to explain how the atomic free bound is determined for row reductions.